presto-parser/src/main/antlr4/com/facebook/presto/sql/parser/SqlBase.g4

```diff
@@ -295,6 +295,8 @@ literal
     | booleanValue                                                                   #booleanLiteral
     | STRING                                                                         #stringLiteral
     | BINARY_LITERAL                                                                 #binaryLiteral
+    | ARRAY '[' (literal (',' literal)*)? ']'                                  #arrayLiteralConstructor
```

Comment:
I don't like that the array/ row construction is duplicated, but I'm not sure how else to only allow literal arrays/rows without removing support for other expressions in arrays elsewhere.   It also still doesn't support maps, since those are created with functions.  Any ideas?